From 5c4e121c17f4e1ae4b583b6b0690c416437beca2 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Sat, 11 Mar 2017 23:04:27 +0800
Subject: [PATCH 1558/5242] MGS-2731 gpu: dts: fix the wrong clock config for
 7ulp gpu shader

commit  7e3c17312839eaeeec185823a7f4c416ba349a2e from
https://source.codeaurora.org/external/imx/linux-imx.git

gc7000nu has only clk2x core input, no external shader clock input,
currently gpu_3d_shader clock is set to the 2d bus clock mistakenly,

should set gpu_3d_shader config with dummy clock in gpu device,
also removed the unused gpu2d_shader_clock setting from dts.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
(cherry picked from commit 021b397e2594a37987d852a382f90bf0559fcf30)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp.dtsi |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index c553c47..0a537c7 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -586,14 +586,13 @@
 				<GIC_SPI 52 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "irq_3d", "irq_2d";
 			clocks = <&clks IMX7ULP_CLK_GPU3D>,
-				<&clks IMX7ULP_CLK_NIC1_DIV>,
+				<&clks IMX7ULP_CLK_DUMMY>,
 				<&clks IMX7ULP_CLK_GPU_DIV>,
 				<&clks IMX7ULP_CLK_GPU2D>,
-				<&clks IMX7ULP_CLK_NIC1_DIV>,
 				<&clks IMX7ULP_CLK_NIC1_DIV>;
 			clock-names = "gpu3d_clk", "gpu3d_shader_clk",
 				"gpu3d_axi_clk", "gpu2d_clk",
-				"gpu2d_shader_clk", "gpu2d_axi_clk";
+				"gpu2d_axi_clk";
 		};
 	};
 
-- 
1.7.9.5

